### DISCUSSION TITLE
Testsuite on custom directory

### DISCUSSION THREAD

**Discussion by underdpt** [pestphp/pest#436] [Q&A]
Hello,

So, I'm developing a plugin for an Laravel app, and would like to add a Pest testsuite to it, but inside the plugin folder to have everything in one place, but am finding a hard way making it work. This is the layout:

```
app folder/
  phpunit.xml
  platform/
    plugins/
      myplugin/
        phpunit.xml
        /tests
          Pest.php, Unit/, Feature/...
  ...
  tests/
    Pest.php, Unit/, Feature/...
```

I installed pest on the main app, and pointed the bootstrap setting on `platform\plugins\myplugin\phpunit.xml` to the main app `autoload.php`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.5/phpunit.xsd"
         bootstrap="../../../vendor/autoload.php"
         cacheResultFile=".phpunit.cache/test-results"
         executionOrder="depends,defects"
         forceCoversAnnotation="true"
         beStrictAboutCoversAnnotation="true"
         beStrictAboutOutputDuringTests="true"
         beStrictAboutTodoAnnotatedTests="true"
         convertDeprecationsToExceptions="true"
         failOnRisky="true"
         failOnWarning="true"
         verbose="true">
```

This works when running

```
 .\vendor\bin\pest -c .\platform\plugins\myplugin\phpunit.xml
```

but it loads `tests/Pest.php` instead of `platform/plugins/myplugin/tests/Pest.php`. I also tried:

```
.\vendor\bin\pest -c .\platform\plugins\myplugin\phpunit.xml --bootstrap .\platform\plugins\myplugin\tests\Pest.php
```

with this it loads both `tests/Pest.php` and `platform/plugins/myplugin/tests/Pest.php`, but that's not the goal.

Also, if it helps, it seems that with this setup the trait `RefreshDatabase` isn't working (using it on both `Pest.php` files, and also on a test class file.

Is this doable or is it a too weird setup that's better to drop it and just use the standard tests folder?

**Comment by gwlortscher** [comment]
Did you get anywhere with this? I prefer putting my tests alongside app code a la React. I know this works without Pest but struggling to get it working with Pest.

**Comment by michaelw85** [comment]
I have a project where my tests folder is not at the root of the project but in some sub directories.
My Pest.php, base class, and helpers would not work. Using the `--test-directory=<DIR>` seems to resolve the issue.

**Comment by shijunti19** [comment]
I also hope to achieve this function
